OloEngine/src/OloEngine/Scene/SceneSerializer.cpp

```
@@ -49,6 +50,243 @@ namespace OloEngine
+            out << YAML::BeginMap;
+            out << YAML::Key << "Time" << YAML::Value << curve.Keys[i].Time;
+            out << YAML::Key << "Value" << YAML::Value << curve.Keys[i].Value;
+            out << YAML::EndMap;
+        }
+        out << YAML::EndSeq;
+        out << YAML::EndMap;
+    }
+
+    static void DeserializeCurve(const YAML::Node& node, ParticleCurve& curve)
+    {
+        if (!node || !node.IsMap())
```

Clang (14.0.6)

[warning] statement should be inside braces

(readability-braces-around-statements)